scripts/install-chef-suse.sh

```diff
@@ -1114,15 +1112,24 @@ for state in "discovering" "discovered" "hardware-installing" \
     "hardware-installed" "installing" "installed" "readying" "ready"
 do
     while [[ -f "/var/run/crowbar/chef-client.lock" ]]; do sleep 1; done
+
     printf "$state: "
     $CROWBAR crowbar transition "$FQDN" "$state" || \
         die "Transition to $state failed!"
+
     if type -f "transition_check_$state"&>/dev/null; then
         "transition_check_$state" || \
             die "Sanity check for transitioning to $state failed!"
     fi
-    # chef_or_die "Chef run for $state transition failed!"
-    chef-client
+
+    if [ "$state" == "hardware-installing" ]; then
+        # Use crowbar_register mode for claiming the disks, as the OS is
+        # already installed
+        echo '{ "crowbar_wall": { "registering": true } }' > $CROWBAR_TMPDIR/chef-registering.json
+        chef-client --json-attributes $CROWBAR_TMPDIR/chef-registering.json
```

Comment:
Well, @MaximilianMeister's version is shorter, and doesn't violate DRY by repeating the same file path twice without a temporary variable (the introduction of which would make your version more correct but even longer).  So in the absence of any concrete counter-arguments, yes I think I prefer his version.  Having said that, I'd be inclined to split the line after the `|` symbol.